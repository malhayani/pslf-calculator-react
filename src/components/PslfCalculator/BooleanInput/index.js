import React from 'react';

const BooleanInput = (props) => {
    return (
        <section className="input-section">
            <label>{props.label}</label>
            <div>
              <label>
                    <label className="boolean-lbl">Yes</label> 
                    { props.value === 'true' 
                        ? <i className="fa fa-check-square fa-fw"></i>
                        : <i className="fa fa-square-o fa-fw" value={true} name={props.name} onClick={props.update}></i>
                    }
                </label>
                {/* If nothing is passed in as a value, don't check either box */}
                <label>
                    <label className="boolean-lbl">No</label>  
                    { props.value === 'true' 
                        ? <i className="fa fa-square-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                        : props.value === 'false' 
                            ? <i className="fa fa-check-square fa-fw"></i>
                            : <i className="fa fa-square-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                    }
                </label>
            </div>
        </section>
    )
}

export default BooleanInput;