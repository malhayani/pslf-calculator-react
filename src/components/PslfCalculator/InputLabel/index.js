import React from 'react';

const InputLabel = (props) => {

    const createLabel = () => {
        return props.valid !== ''
            ? <span>
                <span className="error-lbl">*</span> {props.label} <span className="error-lbl">({props.valid})</span>
            </span>
            : <span>{props.label}</span>;
    }

    return (
        <label className={props.classes}>
        { createLabel() }
        </label>
    )
}

export default InputLabel;