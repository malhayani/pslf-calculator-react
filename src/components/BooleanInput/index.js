import React from 'react';
import InputLabel from '../InputLabel';

const BooleanInput = (props) => {
    return (
        <section className="input-section">
            {
                props.label !== ""
                ? <InputLabel 
                    classes={props.form + "-input-lbl"}
                    label={props.label}
                    valid={props.valid} />
                : ""
            }
            <div>
              <label>
                    <label className="boolean-lbl">Yes</label> 
                    { props.value === 'true' 
                        ? <i className="fa fa-dot-circle-o fa-fw" value={true}></i>
                        : <i className="fa fa-circle-o fa-fw" value={true} name={props.name} onClick={props.update}></i>
                    }
                </label>
                {/* If nothing is passed in as a value, don't check either box */}
                <label>
                    <label className="boolean-lbl">No</label>  
                    { props.value === 'true' 
                        ? <i className="fa fa-circle-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                        : props.value === 'false' 
                            ? <i className="fa fa-dot-circle-o fa-fw" value={false}></i>
                            : <i className="fa fa-circle-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                    }
                </label>
            </div>
        </section>
    )
}

export default BooleanInput;