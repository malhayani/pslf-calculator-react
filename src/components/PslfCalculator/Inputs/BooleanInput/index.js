import React from 'react';
import InputLabel from '../../InputLabel';

const BooleanInput = (props) => {
    return (
        <section className="input-section">
            {
                props.label !== ""
                ? <InputLabel 
                    classes={props.form + "-input-lbl"}
                    label={props.label}
                    valid={props.valid} />
                : ""
            }
            <div>
              <label>
                    <label className="boolean-lbl">Yes</label> 
                    { props.value === 'true' 
                        ? <i className="fa fa-check-square fa-fw"></i>
                        : <i className="fa fa-square-o fa-fw" value={true} name={props.name} onClick={props.update}></i>
                    }
                </label>
                {/* If nothing is passed in as a value, don't check either box */}
                <label>
                    <label className="boolean-lbl">No</label>  
                    { props.value === 'true' 
                        ? <i className="fa fa-square-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                        : props.value === 'false' 
                            ? <i className="fa fa-check-square fa-fw"></i>
                            : <i className="fa fa-square-o fa-fw" value={false} name={props.name} onClick={props.update}></i>
                    }
                </label>
            </div>
        </section>
    )
}

export default BooleanInput;