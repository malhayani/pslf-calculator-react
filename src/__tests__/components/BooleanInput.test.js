import React from 'react';
import BooleanInput from '../../components/BooleanInput';
import InputLabel from '../../components/InputLabel';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

describe('BooleanInput', () => {
    it('renders correctly', () => {
        const wrapper = Enzyme.shallow(<BooleanInput />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Expect no inputs to be selected', () => {
        const wrapper = Enzyme.shallow(<BooleanInput value=""/>);
        expect(wrapper.find('i.fa-dot-circle-o')).toHaveLength(0);
    });

    it('Expect an input to be selected', () => {
        const wrapper = Enzyme.shallow(<BooleanInput value="true"/>);
        expect(wrapper.find('i.fa-dot-circle-o')).toHaveLength(1);
    });

    it('Expect the FALSE input to be selected', () => {
        const wrapper = Enzyme.shallow(<BooleanInput value="false"/>);
        expect(wrapper.find('i.fa-dot-circle-o').at(0).props().value).toEqual(false);
    });

    it('Expect the TRUE input to be selected', () => {
        const wrapper = Enzyme.shallow(<BooleanInput value="true"/>);
        expect(wrapper.find('i.fa-dot-circle-o').at(0).props().value).toEqual(true);
    });

    it('Expect the input to have name prop', () => {
        const wrapper = Enzyme.shallow(<BooleanInput value="true" name="Test"/>);
        expect(wrapper.find('i.fa-circle-o').at(0).props().name).toEqual('Test');
    });

    it('Expect the input to have a label', () => {
        const wrapper = Enzyme.shallow(<BooleanInput label="Test" form="Test" value="true" valid="true" name="Test"/>);
        expect(wrapper.find(InputLabel)).toHaveLength(1);
    });

    it('Expect the input to NOT have a label', () => {
        const wrapper = Enzyme.shallow(<BooleanInput label="" value="true" name="Test"/>);
        expect(wrapper.find(InputLabel)).toHaveLength(0);
    });

    it('Expect button to have been clicked', () => {
        const onClickMock = jest.fn();
        const wrapper = Enzyme.shallow(<BooleanInput value="true" name="Test" update={onClickMock}/>);
        wrapper.find('i.fa-circle-o').simulate('click');

        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});