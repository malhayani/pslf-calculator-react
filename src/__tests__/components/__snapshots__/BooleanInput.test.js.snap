// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanInput renders correctly 1`] = `ShallowWrapper {}`;
